packages/devtools_app/lib/src/timeline/html_timeline_screen.dart

```diff
@@ -430,10 +444,14 @@ class HtmlTimelineScreen extends HtmlScreen {
     // changes the value of [offlineMode], which the button states depend on.
     framework.exitOfflineMode();
 
-    // Revert to the previously selected mode on offline exit. We already
-    // cleared the timeline data - do not repeat this action.
+    // Revert to the previously selected mode on offline exit, unless there is
+    // only one allowed mode for the connected app. We already cleared the
+    // timeline data - do not repeat this action.
+    final mode = await serviceManager.connectedApp.isDartCliApp
```

Comment:
nit: why don't you instead store the initial default value already passed in for the screen and use that here?

Reply:
good idea. done.